src/System.Management.Automation/engine/parser/PSType.cs

```diff
@@ -1119,6 +1119,10 @@ internal static Assembly DefineTypes(Parser parser, Ast rootAst, TypeDefinitionA
                                       .Replace('\\', (char)0x29f9)
```

Comment:
I think it's time to convert this to a loop. This way we have 2 allocations - 1 StringBuilder and 1 string (the final string) instead of possibly 8 strings, and only 1 pass over the filename.